### ISSUE TITLE
In fresh yesod-postgres project, "stack build" always rebuilds certain yesod packages

### ISSUE THREAD

**Issue by kabuhr** [commercialhaskell/stack#2876]
### Summary

Using Stack 1.3.0 to create a fresh yesod-postgres project (with lts-6.27), `stack build` always rebuilds chunks of `yesod`.  It looks like a problem with yesod-persistent thrashing between version 1.4.0.6 and 1.4.1.0. 

### Steps to reproduce

    stack new example yesod-postgres
    cd example
    stack build
    stack build    # oops, rebuilds parts of yesod

Note that, as a workaround, adding `yesod-persistent-1.4.0.6` to `extra-deps` in `stack.yaml` stops the rebuilding.

### Stack version

$ stack --version
Version 1.3.0, Git revision 99b910d3c8e183aa376cb1e6f0341d86aed6d00e (4372 commits) x86_64 hpack-0.15.0

**Comment by snoyberg** [comment]
This is my fault, and not a bug in Stack. I made a mistake in generating the LTS-6.27 snapshot, and accidentally ended up with the wrong cabal file SHA in the snapshot file. When Stack unpacks yesod-persistent-1.4.0.6, it's actually grabbing a .cabal file that says yesod-persistent-1.4.1.0. Then, the next time it runs, it sees that the run version of yesod-persistent is installed, and tries to reinstall.

I'll manually tweak the hash information in the snapshot to work around this, sorry for the confusion, and thanks for the report.

**Comment by snoyberg** [comment]
I've pushed a fix to the lts-haskell repo. You'll need to run the following commands to get the fix in place:

* `rm -f ~/.stack/build-plan/lts-6.27*` to delete the old, incorrect build plan
* `rm -rf ~/.stack/build-plan-cache` to delete the binary cache
* `stack exec -- ghc-pkg unregister yesod-persistent --force` to unregister the incorrect version

From then on, `stack build` _should_ work as expected.

**Comment by kabuhr** [comment]
Thanks.  In following the instructions above, I found that the unregister command produced a warning "unregistering would break the following packages ..." but it was tagged with "(ignoring)", so I assume it worked anyway.  Subsequently, the first time I ran "stack build", it crashed with an error:

    invariant violated: multiple results when describing installed packages

but a second "stack build" completed (and a third "stack build" did *not* attempt to rebuild anything), so the fix seems to have worked.
